File: Retos/Reto #24 - CIFRADO CÉSAR [Fácil]/typescript/Qv1ko.ts
encrypt("Lorem ipsum dolor sit amet.", 6)
decrypt("gjmzh dknph yjgjm ndo vhzo.", 21)

function encrypt(text: string, numberEncrypt: number): void {
    let result = ""
    const letters = text.toLowerCase().split("")
    letters.forEach((letter: string) => {
        result += (letter < "a" || letter > "z")? letter : (letter.charCodeAt(0) + numberEncrypt > 122) ? String.fromCharCode(letter.charCodeAt(0) + numberEncrypt - 26) : String.fromCharCode(letter.charCodeAt(0) + numberEncrypt)
    })
    console.log(result)
}

function decrypt(cesar: string, numberDecrypt: number): void {
    let result = ""
    const letters = cesar.toLowerCase().split("")
    letters.forEach((letter: string) => {
        result += (letter < "a" || letter > "z")? letter : (letter.charCodeAt(0) - numberDecrypt < 97) ? String.fromCharCode(letter.charCodeAt(0) - numberDecrypt + 26) : String.fromCharCode(letter.charCodeAt(0) - numberDecrypt)
    })
    console.log(result)
}

File: Retos/Reto #24 - CIFRADO CÉSAR [Fácil]/java/Qv1ko.java
public class Qv1ko {

    public static void main(String[] args) {
        encrypt("Lorem ipsum dolor sit amet.", 6);
        decrypt("gjmzh dknph yjgjm ndo vhzo.", 21);
    }

    private static void encrypt(String text, int number) {
        String result = "";
        char[] letters = text.toLowerCase().toCharArray();
        for(char letter : letters) {
            result += (char)((letter < 97 || letter > 122)? letter : (letter + number > 122)? letter + number - 26 : letter + number);
        }
        System.out.println(result);
    }

    private static void decrypt(String cesar, int number) {
        String result = "";
        char[] letters = cesar.toLowerCase().toCharArray();
        for(char letter : letters) {
            result += (char)((letter < 97 || letter > 122)? letter : (letter - number < 97)? letter - number + 26 : letter - number);
        }
        System.out.println(result);
    }

}